Update diagrams, top-level schema description and schema reference

duncandewhurst opened the pull request:
**Related issues**

* Closes #178

**Description**

In addition to updating the diagrams per #178, this PR also:

* Updates the top-level description in the schema to align with the diagram in `docs/rdl/what.md`
* Adds the schema descriptions for dataset, resource and each hazard component to `docs/reference/schema.md`

You can preview the changes here: https://rdl-standard.readthedocs.io/en/178-diagrams/reference/schema/

**Merge checklist**

<!-- Complete the checklist before requesting a review. -->

- [x] Update the changelog ([style guide](developer_docs.md#changelog-style-guide))
- [x] Run `./manage.py` pre-commit

If you added or removed a field:

- [ ] Update the `collapse` option of the jsonschema directives for dataset, resource, hazard, exposure, vulnerability and loss on `reference/schema.md`

**Having trouble?**

See [how to resolve check failures](https://github.com/GFDRR/rdl-standard/blob/dev/developer_docs.md#resolve-check-failures).

duncandewhurst commented:
Noting that the exposure diagram may need to be updated if there are further changes in https://github.com/GFDRR/rdl-standard/pull/204

stufraser1 commented on docs/reference/schema.md:
```diff
@@ -10,58 +10,70 @@ This page presents the schema in tables with additional information in paragraph
+  #highlight "publisher" <<required>>
+  #highlight "publisher" / "name" <<required>>
+  #highlight "spatial" <<required>>
+  #highlight "license <<required>>
```
cross-check fields listed as required in diagram and table - e.g. license is required in table, but not highlighted in diagram

duncandewhurst replied:
Ah, good spot. It was in the list of highlighted fields, but was missing a closing `"`. Fixed in https://github.com/GFDRR/rdl-standard/pull/212/commits/cbd6162f3a2b0835b4bf21e855c2ef761afb5a8a.

stufraser1 commented on docs/reference/schema.md:
````diff
@@ -190,40 +227,50 @@ Schema attributes for an earthquake hazard map related to an occurrence probabil
+
+```
+
+The following table lists all exposure component fields:
````
Table still lists metrics as asset cost

duncandewhurst replied:
The tables are generated from the schema, which hadn't changed at the time you reviewed because https://github.com/GFDRR/rdl-standard/pull/204 wasn't yet merged. It is now and I've merged `dev` into this branch so the table has updated auto-magically :-) https://rdl-standard.readthedocs.io/en/178-diagrams/reference/schema/#exposure-metadata

stufraser1 commented on docs/reference/schema.md:
canonical url (line 5) not found when clicked

duncandewhurst replied:
Yep, that URL won't exist until we actually release version 0.2 because it includes the release tag. Once https://github.com/GFDRR/rdl-standard/issues/226 is done we'll also want to update the canonical URLs to use `docs.riskdatalibrary.org` as the base URL rather than `https://raw.githubusercontent.com/GFDRR/rdl-standard/`. I've created an issue about that: https://github.com/GFDRR/rdl-standard/issues/227

duncandewhurst commented:
Based on @matamadio's feedback in yesterday's call, I've removed empty objects and arrays from the diagrams and added an explanatory sentence to describe how arrays are represented.

Here's how the changes look to the hazard diagram (since it is the most complex):

**Before**

![image](https://github.com/GFDRR/rdl-standard/assets/19265814/2ac28052-8e1d-4d12-9457-83eec6266aac)

**After**

![image](https://github.com/GFDRR/rdl-standard/assets/19265814/04b62030-2fd1-46f3-a34d-b0e7d226bef5)

I considered adding an extra object to each array, but I think it makes the diagram too complicated:

![image](https://github.com/GFDRR/rdl-standard/assets/19265814/1deb0ead-8c43-424b-8ec3-94e1c916b6dd)

matamadio reviewed (approved):
Thanks, they read better now.
Just one perplexity on the word "subset". While formally correct, it can induce to think this is a small part of the whole hazard schema; rather, hazard attributes are a subset of the whole schema. But to be aligned with the rest of the docs, I would use "component" instead. Then for each diagram paragraph I would rephrase as:

`The following diagram shows the key attributes of the [hazard] component, with required fields highlighted in blue.`

duncandewhurst commented:
Since @matamadio has approved, I'm going to bypass branch protections and merge this PR so that these changes are available on `dev` before the workshop today. (Merging was blocked because @stufraser1's earlier review was marked as requesting changes, all of which have been addressed).
